src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricsExporter.cs

```diff
@@ -62,75 +63,123 @@ internal DynatraceMetricsExporter(DynatraceExporterOptions options, ILogger<Dyna
+					.GroupBy(x => x.batch)
+					.Select(x => x.Select(v => v.val));
 
-				var metricLines = sb.ToString();
-				_logger.LogDebug(metricLines);
+				ExportResult exportResult;
 
-				// If there are no metric lines, we do not need to send a request.
-				if (metricLines.Length == 0)
+				foreach (var chunk in chunked)
 				{
-					return ExportResult.Success;
-				}
+					var httpRequest = new HttpRequestMessage(HttpMethod.Post, _options.Url);
+					var joinedMetricLines = string.Join(Environment.NewLine, chunk);
 
-				httpRequest.Content = new StringContent(metricLines);
-				try
-				{
-					var response = await _httpClient.SendAsync(httpRequest);
-					if (response.IsSuccessStatusCode)
+					if (joinedMetricLines.Length == 0)
 					{
-						_logger.LogDebug("StatusCode: {StatusCode}", response.StatusCode);
-						exportResults.Add(ExportResult.Success);
+						return ExportResult.Success;
 					}
-					else
+
+					httpRequest.Content = new StringContent(joinedMetricLines);
+					try
 					{
-						_logger.LogError("StatusCode: {StatusCode}", response.StatusCode);
-						_logger.LogError("Content: {Content}", await response.Content.ReadAsStringAsync());
-						exportResults.Add(ExportResult.FailedNotRetryable);
+						// sync over async isn't great, but the exporter is executed in it's own thread.
+						var response = _httpClient.SendAsync(httpRequest).GetAwaiter().GetResult();
+						if (response.IsSuccessStatusCode)
+						{
+							exportResult = ExportResult.Success;
+						}
+						else
+						{
+							_logger.ReceivedErrorResponse(
+								response.StatusCode, response.Content.ReadAsStringAsync().GetAwaiter().GetResult());
+							exportResult = ExportResult.Failure;
+						}
+					}
+					catch (Exception ex)
+					{
+						_logger.FailedSendingMetricLines(ex);
+						exportResult = ExportResult.Failure;
+					}
+
+					if (exportResult == ExportResult.Failure)
+					{
+						// if any batch fails, the entire batch should fail.
+						return exportResult;
 					}
 				}
-				catch (Exception e)
-				{
-					_logger.LogError("Error sending metrics: {Error}", e.Message);
-					throw;
-				}
+
+				return ExportResult.Success;
 			}
+		}
 
-			// if all chunks were exported successfully, return success, otherwise failed.
-			return exportResults.All(x => x == ExportResult.Success) ? ExportResult.Success : ExportResult.FailedNotRetryable;
+		private IReadOnlyCollection<string> GetSerializeMetricLines(in Batch<Metric> batch)
+		{
+			var metricLines = new List<string>();
+
+			foreach (var metric in batch)
+			{
+				metricLines.AddRange(GetSerializedMetricPoints(metric));
+			}
+			return metricLines;
 		}
 
-		private void SerializeMetric(StringBuilder sb, Metric metric)
+		private IReadOnlyCollection<string> GetSerializedMetricPoints(Metric metric)
 		{
-			foreach (var dynatraceMetric in metric.ToDynatraceMetrics(_logger))
+			// TODO: Try to send a PR to get a .Count of the metric points
```

Comment:
Do we have an internal issue for that already? :thinking:

Reply:
No, I'll put on my notes and add a issue on the .NET SDK repo. Then we or someone else could pick it up. I'll remove from the code.